### PR TITLE
refactor: replace misc bluebird API usage

### PR DESCRIPTION
**Pull request by jedwards1211**
- `Promise.method` => (removed, was superfluous)
- `Promise.props` => `p-props` package
- `Promise.promisify`, `Promise.fromCallback` => `require('util').promisify`
- `.timeout` => `p-timeout`
- `Promise.delay` => `delay`

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions?
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### CODE REVIEW

**Comment by jedwards1211** [comment]
merge the `.each` PR first, I'll need to rebase this one and fix conflicts

File: package.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
     "chai-spies": "^1.x",
```
Comment (sushantdhiman) [inline package.json:59]: Missing lock file update, are you using old version for `npm`?

Comment (jedwards1211) [reply package.json:59]: ah sorry, I'm in the habit of using yarn

**Comment by jedwards1211** [comment]
@sushantdhiman thanks for getting these merged fast, and being a great maintainer in general!  I'm going to spend a few days writing a codemod from scratch to unroll the promise chains into async/await.  I've pretty well planned out in my head how it will work and how to avoid [the limitations that [async-await-codemod](https://github.com/sgilroy/async-await-codemod/blob/master/await-promise-chain.js#L239) has.

**Pull request merged**